src/carp.c

```diff
@@ -808,6 +810,17 @@ int docarp(void)
     }
     strncpy(iface.ifr_name, interface, sizeof iface.ifr_name);
 #endif    
+    if (pidfile != NULL) {
+	pid_t pid = getpid();
+	FILE *_pidfile = fopen(pidfile, "w");
+	if (_pidfile != NULL) {
+	    char *buf = malloc(100 * sizeof (long));
```

Comment:
Just to make sure there is enough space for writing the pid to a string. I'm not an every day C programmer. If there is a better way of doing it, please educate me.